tests/conftest.py

```diff
@@ -506,11 +505,11 @@ def isolated_fs(fs):
 
 
 def assert_invoke_exited_with(result: Result, exit_code: int):
-    msg = f"Expected code {exit_code}, got {result.exit_code}"
-    if result.exit_code != exit_code:
-        sys.stderr.write(msg)
-        sys.stdout.write(msg)
-    assert result.exit_code == exit_code, msg
+    msg = f"Expected code {exit_code}, got {result.exit_code}."
+    result_msg = f"\n\n Details of the error: \n {result.stdout}" + (
+        "\n" + result.stderr if result.stderr_bytes is not None else ""
+    )
+    assert result.exit_code == exit_code, msg + result_msg
```

Comment:
Looks good! Maybe you can use a triple-quote string instead of concatenating msg and result_msg?

Also, it would be nice if the message had "stdout:" and "stderr:" prefixes. The message would look like this (this was not in the issue, but I thought about it after trying your branch):

```
Expected code {exit_code}, got {result.exit_code}.

stdout:
{result.stdout}

stderr:
{result.stderr}
```

Reply:
I've changed the format as you requested, thanks for the suggestion!
If every thing seems good with you, I'll squash the last 2 commits into the initial one.